include/FoamAdapter/readers/foamFields.hpp

```diff
@@ -17,31 +17,10 @@ template<typename FoamType>
 auto fromFoamField(const NeoFOAM::Executor& exec, const FoamType& field)
 {
     using type_container_t = typename type_map<FoamType>::container_type;
-    using type_primitive_t = typename type_map<FoamType>::mapped_type;
-    // Create device-side views
-    type_container_t nfField(exec, field.size());
-
-    type_container_t CPUField(NeoFOAM::CPUExecutor {}, field.size());
-    Kokkos::View<type_primitive_t*, Kokkos::HostSpace, Kokkos::MemoryUnmanaged> CPU_view(
-        CPUField.data(), field.size()
+    using mapped_t = typename type_map<FoamType>::mapped_type;
+    type_container_t nfField(
+        exec, reinterpret_cast<const mapped_t*>(field.cdata()), static_cast<size_t>(field.size())
     );
```

Comment:
Why reinterpret cast?

Reply:
cdata can be a pointer to foam::vector, in which case we have to cast foam::vector to neofoam::vector.